core/src/main/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.java

```diff
@@ -358,4 +375,22 @@ public void addMessage(String queue, Message message) {
     public List<TaskExecLog> getTaskExecutionLogs(String taskId) {
         return indexDAO.getTaskExecutionLogs(taskId);
     }
+
+    class DelayWorkflowUpdate implements Runnable {
+        private String workflowId;
+
+        DelayWorkflowUpdate(String workflowId) {
+            this.workflowId = workflowId;
+        }
+
+        @Override
+        public void run() {
+            try {
+                Workflow workflow = executionDAO.getWorkflow(workflowId, false);
```

Comment:
Hi @apanicker-nflx , I just wanted to tell that in high throughput scenario, we will not have data in Redis also. We remove it the moment it gets complete. What we can do here is have data while creating  DelayWorkflowUpdate object and schedule it to run after 30 seconds.

Reply:
>in high throughput scenario, we will not have data in Redis also. We remove it the moment it gets complete.

In this case, are you removing the workflow synchronously when workflow enters terminal state? I would suggest using the [searchArchivableWorkflows](https://github.com/Netflix/conductor/blob/master/es5-persistence/src/main/java/com/netflix/conductor/dao/es5/index/ElasticSearchDAOV5.java#L639) method in ES to remove the workflows using a background process. 

> What we can do here is have data while creating DelayWorkflowUpdate object and schedule it to run after 30 seconds.

I intentionally am passing `workflowId` only to `DelayWorkflowUpdate` object so as to reduce the memory footprint in this case. Since the `ScheduledThreadPoolExecutor` uses an unbounded queue, if the number of short running workflows are too large this would avoid having to store the entire workflow object with each Runnable.